### ISSUE TITLE
Fix S2479 FP: Ignore raw string literals

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6292]
### Description

FN for S2479 (ControlCharacterInString).

Using C#11 multi-line string interpolation, the rule is not raising an issue.

### Repro steps

```csharp
public const string RawStringLiteralsNonCompliant = """test"""; // Noncompliant

void NewlinesInStringInterpolation()
{ 
    string NewlinesInterpolatedStringNonCompliant = $"test{RawStringLiteralsNonCompliant +
        RawStringLiteralsNonCompliant}"; // FN
    string NewlinesInterpolatedStringRawNonCompliant = $$"""test{{RawStringLiteralsNonCompliant +
        RawStringLiteralsNonCompliant}}"""; // FN
}

```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

**Comment by gregory-paidis-sonarsource** [comment]
This can be marked as resolved after PR #6345 gets merged.

For more details, read [this comment](https://github.com/SonarSource/sonar-dotnet/pull/6345#issuecomment-1310411544)

**Comment by gregory-paidis-sonarsource** [comment]
After internal discussion with @csaba-sagi-sonarsource , we decided to ignore multi-line interpolation in the case of raw and/or verbatim string literals, since they do not have an escape mechanism. 

Examples:

```csharp
        void NewlinesInStringInterpolation()
        {
            var compliant = "test"; // Compliant
            var nonCompliant = "test"; // Noncompliant

            var baseCase = $"test{
                compliant
                }"; // Compliant

            var interpolatedTextHasControlCharacter = $"test{
                nonCompliant
                }"; // Compliant, interpolated text ignored

            var normalTextHasControlCharacter = $"test{
                nonCompliant
                }"; // Noncompliant@-2

            var verbatimWithControlCharacter = @$"test{
                nonCompliant
                }"; // Compliant, verbatim

            var rawWithControlCharacter = $"""test{
                nonCompliant
                }"""; // Compliant, raw

            var rawVerbatimWithControlCharacter = @$"""test{
                nonCompliant
                }"""; // Compliant, raw+verbatims
        }


```
